src/stronghold/mod.rs

```diff
@@ -76,7 +78,7 @@ use crate::{Error, Result};
 #[builder(pattern = "owned", build_fn(skip))]
 pub struct StrongholdAdapter {
     /// A stronghold instance.
-    stronghold: Stronghold,
+    stronghold: Arc<Mutex<Stronghold>>,
```

Comment:
Wouldn't a `RwLock` be better here?

Reply:
Probably doesn't really help

Reply:
Maybe, but only `SecretManager` has 2 methods taking `&self`, while others take `&mut self`